app/src/main/java/org/thoughtcrime/securesms/database/MmsSmsDatabase.java

```diff
@@ -96,6 +99,14 @@ public MmsSmsDatabase(Context context, SQLCipherOpenHelper databaseHelper) {
     }
   }
 
+  public @Nullable MessageRecord getNonDeletedMessageForTimestamp(long timestamp) {
```

Comment:
Isn't a non-deleted message just a message? Perhaps `getMessageForTimestamp`?

Ah, we now have messages with a 'deleted' type in the database.... that's a kinda odd way to go about things, but another way you could think of it is that the message is kinda in the "Recycle Bin" / "Trash" whatever you want to call it, so could potentially be recovered if required (but we never do and just clean that stuff out).

So okay, I guess...

Reply:
Yeah a message can be completely gone or marked as deleted. It still exists in the db and shows up as a special case message with the bin icon and a label like 'This message was deleted for everyone'